### PR TITLE
Channelify zero conf test

### PR DESCRIPTION
**Pull request by connorwstein**

### CODE REVIEW

**Review by archseer — DISMISSED** [review]

File: core/services/log/broadcaster_internal_test.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+			BlockNumber: 2,
+			Index:       2,
+			Topics: []common.Hash{
+				(flux_aggregator_wrapper.FluxAggregatorNewRound{}).Topic(),
+				utils.NewHash(),
+				utils.NewHash(),
+			},
+			Data: []byte("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"),
+		},
+	}
```
Comment (PiotrTrzpil) [inline core/services/log/broadcaster_internal_test.go:127]: could use `cltest.RawNewRoundLog` for setting up these

File: core/services/log/broadcaster_internal_test.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+		case <-time.After(time.Second):
+			t.Error("failed to send log to log broadcaster")
+		}
+	}
+	for i := 0; i < 10; i++ {
+		if len(lb.logPool.logsByBlockHash[bh]) == len(addr1SentLogs) {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
```
Comment (PiotrTrzpil) [inline core/services/log/broadcaster_internal_test.go:160]: i think it's a rather typical case to use gomega.Eventually ?

File: core/services/log/broadcaster_internal_test.go (hunk 1 above)
Comment (connorwstein) [reply core/services/log/broadcaster_internal_test.go:122]: cltest creates an import cycle. I'm general I think it should only be used for integration level tests

**Comment by connorwstein** [comment]
> I like it! though might be good to keep both tests? The flakiness of the deleted one is because sometimes the logs are sent on head 5 instead of head 4, will look into that

The logs come in at different heads because of SimulateIncomingHeads - this internal test waits for them to appear in the pool then just sends one head to fire them all off. Then its guaranteed they come in on the same head

File: core/services/log/pool.go [hunk 3]
```diff
@@ -9,13 +9,17 @@ import (
...
+	// A mapping of block numbers to a set of block hashes for all
+	// the logs in the pool.
+	hashesByBlockNumbers map[uint64]map[common.Hash]struct{}
+	// A mapping of blockhashes to logs
+	logsByBlockHash map[common.Hash][]types.Log
+	// This min-heap maintains block numbers of logs in the pool.
+	// it helps us easily determine the minimum log block number
+	// in the pool (while the set of log block numbers is dynamically changing).
+	heap *pairingHeap.PairHeap
+}
```
Comment (PiotrTrzpil) [inline core/services/log/pool.go:22]: ah great thought to add comments 👍🏼

**Review by PiotrTrzpil — APPROVED** [review]

**Pull request merged**